services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.spec.ts

```diff
@@ -0,0 +1,342 @@
+
+    await expect(service.get(mockAppDocument.uuid)).rejects.toMatchObject(
+      new ServiceNotFoundException(
+        `Failed to find document ${mockAppDocument.uuid}`,
+      ),
+    );
+  });
+
+  it('should call through for list', async () => {
+    const mockDocument = {};
+    const mockNoiDocument = {
+      uuid: '1',
+      document: mockDocument,
+    } as NoticeOfIntentDocument;
+    mockRepository.find.mockResolvedValue([mockNoiDocument]);
+
+    const res = await service.list(fileNumber);
+
+    expect(mockRepository.find).toHaveBeenCalledTimes(1);
+    expect(res[0]).toBe(mockNoiDocument);
+  });
+
+  it('should call through for download', async () => {
+    const mockDocument = {};
+    const mockAppDocument = {
+      uuid: '1',
+      document: mockDocument,
+    } as NoticeOfIntentDocument;
+
+    const fakeUrl = 'mock-url';
+    mockDocumentService.getDownloadUrl.mockResolvedValue(fakeUrl);
+
+    const res = await service.getInlineUrl(mockAppDocument);
+
+    expect(mockDocumentService.getDownloadUrl).toHaveBeenCalledTimes(1);
+    expect(res).toEqual(fakeUrl);
+  });
+
+  it('should load all applicant sourced documents correctly', async () => {
+    const mockAppDocument = new NoticeOfIntentDocument({
+      uuid: '1',
+      document: new Document({
+        source: DOCUMENT_SOURCE.APPLICANT,
+      }),
+    });
+    const mockLgDocument = new NoticeOfIntentDocument({
+      uuid: '2',
+      document: new Document({
+        source: DOCUMENT_SOURCE.LFNG,
+      }),
+    });
+
+    mockRepository.find.mockResolvedValue([mockAppDocument, mockLgDocument]);
+
+    const res = await service.getApplicantDocuments('1');
+
+    expect(mockRepository.find).toHaveBeenCalledTimes(1);
+    expect(res.length).toEqual(1);
+    expect(res[0]).toBe(mockAppDocument);
+  });
+
+  it('should call delete for each document loaded', async () => {
+    const mockAppDocument = new NoticeOfIntentDocument({
+      uuid: '1',
+      document: new Document({
+        source: DOCUMENT_SOURCE.APPLICANT,
+      }),
+    });
+    const mockLgDocument = new NoticeOfIntentDocument({
+      uuid: '2',
+      document: new Document({
+        source: DOCUMENT_SOURCE.LFNG,
+      }),
+    });
+
+    mockRepository.find.mockResolvedValue([mockAppDocument, mockLgDocument]);
+    mockDocumentService.softRemove.mockResolvedValue();
+    mockRepository.remove.mockResolvedValue({} as any);
+
+    const res = await service.deleteByType(DOCUMENT_TYPE.STAFF_REPORT, '');
+
+    expect(mockRepository.find).toHaveBeenCalledTimes(1);
+    expect(mockDocumentService.softRemove).toHaveBeenCalledTimes(2);
+  });
+
+  it('should call through for fetchTypes', async () => {
+    mockTypeRepository.find.mockResolvedValue([]);
+
+    const res = await service.fetchTypes();
+
+    expect(mockTypeRepository.find).toHaveBeenCalledTimes(1);
+    expect(res).toBeDefined();
+  });
+
+  // it('should set the type and description for multiple files', async () => {
```

Comment:
I assume this one is also going to be enabled later?

Reply:
yes, added a comment specifying this is for Step 7